include/deal.II/base/index_set.h

```diff
@@ -1683,6 +1681,17 @@ IndexSet::index_within_set (const size_type n) const
   p = Utilities::lower_bound(range_begin, range_end, r,
                              Range::end_compare);
 
+  // if we did not find, then
+  if (p==range_end)
```

Comment:
put both into a single ``if`` and replace the comment by something like "if n is not in this set, return failure"